plugins/generator-1.17.1/forge-1.17.1/templates/biome/tree_leave_decorator.java.ftl

```diff
@@ -0,0 +1,99 @@
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+package ${package}.world.level.levelgen.feature.treedecorators;
+<#include "../mcitems.ftl">
+
+public class ${name}LeaveDecorator extends LeaveVineDecorator {
+
+        public static final ${name}LeaveDecorator instance = new ${name}LeaveDecorator();
+        public static com.mojang.serialization.Codec<LeaveVineDecorator> codec;
+        public static TreeDecoratorType<?> tdt;
+
+        static {
+            codec = com.mojang.serialization.Codec.unit(() -> instance);
+            tdt = new TreeDecoratorType<>(codec);
+            tdt.setRegistryName("${registryname}_ld");
+            ForgeRegistries.TREE_DECORATOR_TYPES.register(tdt);
+        }
+
+        @Override
+        protected TreeDecoratorType<?> type() {
+            return tdt;
+        }
+
+        @Override
+        public void place(LevelSimulatedReader level, BiConsumer<BlockPos, BlockState> biConsumer, Random p_161737_, List<BlockPos> p_161738_, List<BlockPos> p_161739_) {
+            p_161739_.forEach((blockpos) -> {
+                if (p_161737_.nextInt(4) == 0) {
+                    BlockPos bp = blockpos.west();
+                    if (Feature.isAir(level, bp)) {
+                        addVine(level, bp, biConsumer);
+                    }
+                }
+
+                if (p_161737_.nextInt(4) == 0) {
+                    BlockPos blockpos1 = blockpos.east();
+                    if (Feature.isAir(level, blockpos1)) {
+                        addVine(level, blockpos1, biConsumer);
+                    }
+                }
+
+                if (p_161737_.nextInt(4) == 0) {
+                    BlockPos blockpos2 = blockpos.north();
+                    if (Feature.isAir(level, blockpos2)) {
+                        addVine(level, blockpos2, biConsumer);
+                    }
+                }
+
+                if (p_161737_.nextInt(4) == 0) {
+                    BlockPos blockpos3 = blockpos.south();
+                    if (Feature.isAir(level, blockpos3)) {
+                        addVine(level, blockpos3, biConsumer);
+                    }
+                }
+
+            });
+        }
+
+        private static void addVine(LevelSimulatedReader p_161730_, BlockPos p_161731_, BiConsumer<BlockPos, BlockState> p_161733_) {

```

Comment:
Consider refactoring these unmapped fields to reasable names where possible to make the code nicer